apps/meteor/app/authentication/server/startup/index.js

```diff
@@ -40,7 +40,10 @@ Accounts.config({
  *
  * we are removing the status here because meteor send 'offline'
  */
-Object.assign(Accounts._defaultPublishFields.projection, (({ status, ...rest }) => rest)(getBaseUserFields()));
+Object.assign(
+	Accounts._defaultPublishFields.projection,
+	(({ status, ...rest }) => rest)({ ...getBaseUserFields(), 'services.totp.enabled': 1, 'services.email2fa.enabled': 1 }),
```

Comment:
Why not simply add the two fields to `getBaseUserFields` directly?

The only other place that uses this function already adds the whole `service` to it anyway. That other place could even be using `getDefaultUserFields` instead.

Reply:
Ideally, we should be publishing the exact same fields as the `api/v1/me` endpoint returns;

The problem is that that place returns calculated data too (for example, it creates `services.password.exists` based on `services.password.bcrypt`), which is beyond what we can do with a simple projection list; We would need to override some meteor code to be able to calculate it for the publication too.

Reply:
> Why not simply add the two fields to getBaseUserFields directly?

Because we use this helper on `me` endpoint, and there we need the entire `services` object. If we make it part of helper, the mongo query would error out due to overlapping projection fields.


> That other place could even be using getDefaultUserFields instead.

Not really, because we need the entire `services` obj to check if user is OAuth user or not.

But I do see your point, this entire thing needs some refactor and planning.